include/fmt/core.h

```diff
@@ -902,20 +909,37 @@ class basic_parse_context : private ErrorHandler {
   FMT_CONSTEXPR unsigned next_arg_id();
 
   FMT_CONSTEXPR bool check_arg_id(unsigned) {
-    if (next_arg_id_ > 0) {
+    if (is_indexing_arguments_automatically()) {
```

Comment:
Changes to indexing seem unrelated, please submit as a separate PR to make this one manageable.

Reply:
It was introduced to simplify the `bool check_arg_id(internal::auto_id)` implementation. I'll revert it (see below).